.github/workflows/steam.yml

```diff
@@ -0,0 +1,212 @@
+  STEAM_STABLE_BRANCH: staging
+  STEAM_BETA_BRANCH: beta_staging
+
+jobs:
+  upload:
+    name: Steam upload
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        path: source
+
+    - name: Setup 7-zip (PPA)
+      # The 7zip version available in the default ubuntu repos (p7zip) is wildly out-of-date and does not properly support DMG files
+      run: |
+        sudo add-apt-repository -y ppa:spvkgn/sevenzip
+        sudo apt update -y
+        sudo apt install -y 7-zip
+
+    - name: Get build information
+      run: |
+        EVENT=${{ github.event_name }}
+        if [[ $EVENT == "release" || ( $EVENT == "workflow_dispatch" && ! -z "${{ github.event.inputs.tag }}") ]]; then
+            if [[ $EVENT == "release" ]]; then
+                DESC=${{ github.event.release.tag_name }}
+                if [ "${{ github.event.release.prerelease }}" = "true" ]; then
+                  BRANCH=${{ env.STEAM_BETA_BRANCH }}
+                else
+                  BRANCH=${{ env.STEAM_STABLE_BRANCH }}
+                fi
+                ASSETS_URL=${{ github.event.release.assets_url }}
+            else
+                RELEASE=$(curl -s "${{ github.api_url }}/repos/obsproject/obs-studio/releases/tags/${{ github.event.inputs.tag }}")
+                
+                DESC=$(jq -r '.tag_name' <<< "$RELEASE")
+                if [ $(jq -r '.prerelease' <<< "$RELEASE") = "true" ]; then
+                  BRANCH=${{ env.STEAM_BETA_BRANCH }}
+                else
+                  BRANCH=${{ env.STEAM_STABLE_BRANCH }}
+                fi
+                ASSETS_URL=$(jq -r '.assets_url' <<< "$RELEASE")
+            fi
+
+            ASSETS=$(curl -s "$ASSETS_URL")
+            WIN_ASSET_URL=$(jq -r '.[] | select(.name|test(".*x64.zip")) .browser_download_url' <<< "$ASSETS")
+            MAC_ASSET_URL=$(jq -r '.[] | select(.name|test(".*.dmg")) .browser_download_url' <<< "$ASSETS")
+            TYPE="release"
+        else
+            BRANCH=${{ env.STEAM_NIGHTLY_BRANCH }}
+            BUILDS=$(curl -s "${{ github.api_url }}/repos/obsproject/obs-studio/actions/workflows/${{ env.WORKFLOW_ID }}/runs?per_page=1&event=push&status=success&branch=${{ env.GIT_NIGHTLY_BRANCH }}")
+            ARTIFACTS_URL=$(jq -r '.workflow_runs[].artifacts_url' <<< "$BUILDS")
+            DESC="g$(jq -r '.workflow_runs[].head_sha' <<< "$BUILDS" | cut -c1-9)"
+
+            ARTIFACTS=$(curl -s $ARTIFACTS_URL)
+            WIN_ASSET_URL=$(jq -r '.artifacts[] | select(.name|test(".*win64")) .archive_download_url' <<< "$ARTIFACTS")
+            MAC_ASSET_URL=$(jq -r '.artifacts[] | select(.name|test(".*macos-x86_64")) .archive_download_url' <<< "$ARTIFACTS")
+            TYPE="nightly"
+        fi
+        
+        # Apply overrides from workflow_dispatch
+        if [[ $EVENT == "workflow_dispatch" ]]; then
+            if [[ ! -z "${{ github.event.inputs.win_url_override }}" ]]; then
```

Comment:
I'm not sure whether `-n` might suffice here - testing that only returns true if the string is set _and_ is not zero-length.